src/core/Akka.Cluster/ClusterDaemon.cs

```diff
@@ -1996,27 +2004,29 @@ public GossipStats IncrementOlderCount()
     /// <summary>
     /// INTERNAL API
     /// 
-    /// The supplied callback will be run once when the current cluster member is <see cref="MemberStatus.Up"/>
+    /// The supplied callback will be run once when the current cluster member has the same status.
     /// </summary>
-    class OnMemberUpListener : ReceiveActor
+    internal class OnMemberStatusChangedListener : ReceiveActor
```

Comment:
Refactored this actor to handle any arbitrary `MemberStatus` now - this actor gets re-used for both `RegisterOnMemberUp` and `RegisterOnMemberRemoved` now